cfgrammar/src/lib/yacc/reporting.rs

```diff
@@ -44,6 +45,11 @@ pub trait ErrorReport {
     fn finish(&mut self) {}
 }
 
+pub trait ErrorFormatter {
+    fn format_error(&self, e: &YaccGrammarError) -> Box<dyn Error>;
```

Comment:
One thing we could consider (Which I haven't tried so it could be a bad idea) is if something like `Into<Box<dyn Error>>` would be better?